src/shell/shell_cmds.cpp

```diff
@@ -2532,6 +2492,98 @@ static std::vector<std::string> search_files(const std::string_view query)
 	return files;
 }
 
+struct FileExt {
+	std::string root      = {};
+	std::string extension = {};
+};
+
+static FileExt split_extention(const std::string& fullname)
+{
+	FileExt ret = {};
+	size_t pos  = fullname.rfind('.');
+	ret.root    = fullname.substr(0, pos);
+	++pos;
+	if (pos > 0 && pos < fullname.size()) {
+		ret.extension = fullname.substr(pos);
+	} else {
+		ret.extension = {};
+	}
+	return ret;
```

Comment:
Can't we use std::filesystem functions for this? Or we want to make sure we make something very DOS specific here?

Reply:
I think it's best to avoid `std::filesystem` when interacting with the DOS side of things.  As I understand it, those functions are meant for abstracting host side differences (ex. on Unix-like systems it may replace DOS-style path separators `\` with Unix-style `/` which we most certainly do not want).

Reply:
I think you-re right, it would be asking for trouble.